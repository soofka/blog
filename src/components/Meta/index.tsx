import * as React from 'react';
import { Helmet } from 'react-helmet';

export interface MetaPropsInterface {
  title?: string;
  subtitle?: string;
  description?: string;
  keywords?: string;
  imageUrl?: string;
}

export const Meta = (props: MetaPropsInterface) => {
  const { title, subtitle, description, keywords, imageUrl } = props;

  const metaElements = [];

  if (title) {
    metaElements.push(
      <title key="title">{title}</title>,
      <meta key="og:site_name" property="og:site_name" content={title} />,
      <meta key="twitter:site" property="twitter:site" content={title} />,
      subtitle
        ? <meta key="twitter:title" property="twitter:title" content={subtitle} />
        : <meta key="twitter:title" property="twitter:title" content={title} />,
    );
  }

  if (description) {
    metaElements.push(
      <meta key="description" name="description" content={description} />,
      <meta key="og:description" property="og:description" content={description} />,
      <meta key="twitter:description" property="twitter:description" content={description} />,
    );
  }

  if (keywords) {
    metaElements.push(
      <meta key="keywords" name="keywords" content={keywords} />,
    );
  }

  if (imageUrl) {
    metaElements.push(
      <meta key="og:image" name="og:image" content={imageUrl} />,
      <meta key="twitter:image" name="twitter:image" content={imageUrl} />,
    );
  }

  return(
    <Helmet>
      {metaElements}
    </Helmet>
  );
};

export default Meta;
