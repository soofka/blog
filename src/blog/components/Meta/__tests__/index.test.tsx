import * as React from 'react';
import { shallow } from 'enzyme';

import Meta from 'components/Meta';

describe('<Meta/>', () => {

  it('renders correctly', () => {
    expect(shallow(
      <Meta/>,
    )).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    expect(shallow(
      <Meta
        title="testTitle"
        subtitle="testSubtitle"
        description="testDescription"
        keywords="testKeywords"
        imageUrl="testImageUrl"
      />,
    )).toMatchSnapshot();
  });

});
