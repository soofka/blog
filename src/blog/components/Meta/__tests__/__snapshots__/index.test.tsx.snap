// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta/> renders correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
/>
`;

exports[`<Meta/> renders correctly with optional props 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title
    key="title"
  >
    testTitle
  </title>
  <meta
    content="testTitle"
    key="og:site_name"
    property="og:site_name"
  />
  <meta
    content="testTitle"
    key="twitter:site"
    property="twitter:site"
  />
  <meta
    content="testSubtitle"
    key="twitter:title"
    property="twitter:title"
  />
  <meta
    content="testDescription"
    key="description"
    name="description"
  />
  <meta
    content="testDescription"
    key="og:description"
    property="og:description"
  />
  <meta
    content="testDescription"
    key="twitter:description"
    property="twitter:description"
  />
  <meta
    content="testKeywords"
    key="keywords"
    name="keywords"
  />
  <meta
    content="testImageUrl"
    key="og:image"
    name="og:image"
  />
  <meta
    content="testImageUrl"
    key="twitter:image"
    name="twitter:image"
  />
</HelmetWrapper>
`;
